tests/validators/test_typed_dict.py

```diff
@@ -336,6 +354,52 @@ def test_validate_assignment_allow_extra_validate():
     ]
 
 
+def test_validate_assignment_with_strict():
+    v = SchemaValidator(
+        {'type': 'typed-dict', 'fields': {'x': {'schema': {'type': 'str'}}, 'y': {'schema': {'type': 'int'}}}}
+    )
+
+    r = v.validate_python({'x': 'a', 'y': '123'})
+    assert r == {'x': 'a', 'y': 123}
+
+    assert v.validate_assignment('y', '124', r) == {'x': 'a', 'y': 124}
+
+    with pytest.raises(ValidationError) as exc_info:
+        v.validate_assignment('y', '124', r, True)
+
+    assert exc_info.value.errors() == [
+        {'kind': 'int_type', 'loc': ['y'], 'message': 'Input should be a valid integer', 'input_value': '124'}
+    ]
+
+
+# TODO: move to tests/test_validation_context.py
+# once https://github.com/samuelcolvin/pydantic-core/pull/220 is merged
```

Comment:
@samuelcolvin If you're ok with this PR, I'll rebase it once https://github.com/samuelcolvin/pydantic-core/pull/220 is merged and remove those two lines

Reply:
Why don't we just merge #220? We can add more later.

Reply:
I don't understand your remark 🤔 Yes let's merge #220

Reply:
I mean we can add more tests if required later. Merging now.

Reply:
Rebased